crt_portal/cts_forms/filters.py

```diff
@@ -61,12 +61,16 @@ def report_filter(request):
             elif filter_options[field] == '__contains':
                 kwargs[f'{field}__icontains'] = request.GET.getlist(field)[0]
             elif 'date' in field:
-                # filters by a start date or an end date expects YYYYMMDD
+                # filters by a start date or an end date expects yyyy-mm-dd
                 field_name = _get_date_field_from_param(field)
-                year = int(request.GET.getlist(field)[0][:4])
-                month = int(request.GET.getlist(field)[0][4:6])
-                day = int(request.GET.getlist(field)[0][6:])
-                kwargs[f'{field_name}{filter_options[field]}'] = datetime.date(year, month, day)
+                encodedDate = request.GET.getlist(field)[0]
+                decodedDate = urllib.parse.unquote(encodedDate)
+                try:
+                    dateObj = datetime.strptime(decodedDate, "%Y-%m-%d")
+                except ValueError:
+                    # if the date is invalid, set it to 2020 because there is no data prior to 2020
+                    dateObj = datetime.strptime("2020", "%Y")
```

Comment:
:sweat_smile: I am still not comfortable with this. Can we `continue` instead if we hit an exception? The reason I keep asking about this is because this is not intuitive behavior. 

If I filter by `create_date_start=2020-07-28234` for instance, this appears to work (and filters >= 2020 reports). Conversely, if I filter by `create_date_end=2020-07-28234` then nothing shows up (since it's still filtering <= 2020 reports). This isn't right -- I as the user have no indication as to why this is happening -- what's the difference between filtering start and end with invalid input, if I didn't know the codebase? 

If error handling for user input is not in scope, that's fine. Can we ignore invalid input -- at least for now? 

For instance, `status=improved` works -- it still filters by that field, but it works. Ideally I would like more robust error handling but I realize we're not really set up for that.

Reply:
error handling and displaying is not in scope for sure, there is no framework for displaying errors in the filters now. For now ignoring is probably okay for now.
Where can I get more information on 'status=improved'? don't know about it.

Reply:
Logging invalid dates and ignoring from filtering, with this code change. Lmk your thoughts. 

                 decodedDate = urllib.parse.unquote(encodedDate)
                 try:
                     dateObj = datetime.strptime(decodedDate, "%Y-%m-%d")
+                    kwargs[f'{field_name}{filter_options[field]}'] = dateObj
                 except ValueError:
-                    # if the date is invalid, set it to 2020 because there is no data prior to 2020
-                    dateObj = datetime.strptime("2020", "%Y")
-                kwargs[f'{field_name}{filter_options[field]}'] = dateObj
+                    # if the date is invalid, we ignore it.
+                    logger.info("Invalid date format specified for " + f'{field_name}{filter_options[field]}')

Reply:
`status=improved` was an example, I put down invalid input for that field to see what it would do.

Can we do something like:

```python
    decodedDate = urllib.parse.unquote(encodedDate)
    try:
        dateObj = datetime.strptime(decodedDate, "%Y-%m-%d")
        kwargs[f'{field_name}{filter_options[field]}'] = dateObj
    except ValueError:
        continue
```

This effectively ignores invalid input. Just a thought, try it out and see what happens?

Reply:
Done..